src/braket_container.py

```diff
@@ -254,41 +225,74 @@ def install_additional_requirements() -> None:
+    if entry_point.find(":") >= 0:
+        str_module, _, str_method = entry_point.partition(":")
+        customer_module = importlib.import_module(str_module)
+        customer_method = getattr(customer_module, str_method)
+    else:
+        def customer_method():
+            return runpy.run_module(entry_point, run_name="__main__")
+    return customer_method
+
+
+def wrap_customer_method(customer_method: Callable) -> Callable:
+    def wrapped_customer_method(**kwargs):
+        try:
+            return customer_method(**kwargs)
+        except Exception as e:
+            exception_type = type(e).__name__
+            exception_string = (
+                exception_type
+                if not e.args
+                else f"{exception_type}: {e}"
+            )
+            _log_failure(exception_string, display=False)
+            raise e
+    return wrapped_customer_method
+
+
+def kick_off_customer_script(customer_method: Callable) -> multiprocessing.Process:
+    """
+    Runs the customer script as a separate process.
 
     Args:
-        entry_point (str): the code to run in the format <package>:<method>.
+        customer_method (Callable): The customer method to be run.
 
     Returns:
-        int: The exit code of the customer code run.
+        Process: the process handle to the running process.
     """
     print("Running Code As Process")
-    customer_code_process = kick_off_customer_script(entry_point)
-    join_customer_script(customer_code_process)
-    print("Code Run Finished")
-    return customer_code_process.exitcode
+    wrapped_customer_method = wrap_customer_method(customer_method)
+    process_kwargs = {"target": wrapped_customer_method}
 
+    function_args = try_bind_hyperparameters_to_customer_method(customer_method)
+    if function_args is not None:
+        process_kwargs["kwargs"] = function_args
 
-def run_customer_code_as_subprocess(entry_point: str) -> int:
+    customer_code_process = multiprocessing.Process(**process_kwargs)
+    customer_code_process.start()
+    return customer_code_process
+
+
+def join_customer_script(customer_code_process: multiprocessing.Process):
     """
-    When provided just the name of the module to run, we run it as a subprocess.
+    Joins the process running the customer code.
 
     Args:
-        entry_point (str): the name of the module to run.
-
-    Returns:
-        int: The exit code of the customer code run.
+        customer_code_process (Process): the process running the customer code.
     """
-    print("Running Code As Subprocess")
     try:
-        result = subprocess.run(["python", "-m", entry_point], cwd=EXTRACTED_CUSTOMER_CODE_PATH)
+        customer_code_process.join()
     except Exception as e:
-        log_failure_and_exit(f"Unable to run job at entry point {entry_point}\nException: {e}")
+        log_failure_and_exit(f"Job did not exit gracefully.\nException: {e}")
```

Comment:
Should we standardize how we print this exception as well?

Reply:
I don't have much context here, but it seems like this codepath is hard to reach. Do you know what causes a join to fail here?

Reply:
It would most likely be OS issues that would cause it. It doesn't hurt to be safe to catch it.